File: ClinicSystem/src/com/example/clinicsystem/models/classes/M_UserType.java
package com.example.clinicsystem.models.classes;

import com.example.clinicsystem.helpers.DatabaseConnection;

import java.util.ArrayList;
import java.util.List;

public class M_UserType {
    private int userTypeID;
    private String typeName;
    private List<M_Permission> permissions;

    public M_UserType(int userTypeID, String typeName,List<M_Permission> permissions) {
        this.userTypeID = userTypeID;
        this.typeName = typeName;
        this.permissions = permissions;
    }

    public M_UserType() {
        permissions = new ArrayList<>();
    }

    public int addUserType() {
        return DatabaseConnection.getINSTANCE().insertInto("usertype","(UserType_Name)"
                ,"('" + this.getTypeName() + "')");
    }

    public void addTypeLink() {
        for(M_Permission permission : this.permissions) {
            DatabaseConnection.getINSTANCE().insertInto("type_permission","(Permission_ID," +
                    " UserType_ID)", "('" + permission.getPermissionID() + "', '" + this.getUserTypeID() + "')");
        }
    }

    public int getUserTypeID() {
        return userTypeID;
    }

    public void setUserTypeID(int userTypeID) {
        this.userTypeID = userTypeID;
    }

    public String getTypeName() {
        return typeName;
    }

    public void setTypeName(String typeName) {
        this.typeName = typeName;
    }

    public List<M_Permission> getPermissions() {
        return permissions;
    }

    public void setPermissions(List<M_Permission> permissions) {
        this.permissions = permissions;
    }
}

File: clinic/M_UserType.php
<?php
include_once "M_Permission.php";

class M_UserType {
	private $userTypeID;
    private $typeName;
    private $permissions;

    public function __construct($userTypeID, $typeName) {
        $this->userTypeID =  $userTypeID;
        $this->typeName =    $typeName;
        //$this->permissions = $permissions;
    }

    public function addUserType() {
        $conn = DatabaseConnection::getINSTANCE();
		
		return $conn->insertInto("usertype", "(UserType_Name)", "('". $this->typeName . "')");
    }

    public function addTypeLink() {
		$conn = DatabaseConnection::getINSTANCE();
		
        foreach($this->permissions as $permission) {
            $conn->insertInto("type_permission", "(Permission_ID, UserType_ID)",
			"('" . $permission->getPermissionID() . "', '" . $this->userTypeID . "')");
        }
    }

	public function getAllUserTypes($condition) {
        $conn = DatabaseConnection::getINSTANCE();
		
		$result = $conn->select("usertype", $condition);
		$usertypes = array();
		
		while ($row = $result->fetch_assoc()) {
			$usertype = new M_UserType($row["UserType_ID"], $row["UserType_Name"]);
			array_push($usertypes, $usertype);
		}
		
		return $usertypes;
    }
	
    public function getUserTypeID() {
        return $this->userTypeID;
    }

    public function setUserTypeID($userTypeID) {
        $this->userTypeID = $userTypeID;
    }

    public function getTypeName() {
        return $this->typeName;
    }

    public function setTypeName($typeName) {
        $this->typeName = $typeName;
    }

    public function getPermissions() {
        return $this->permissions;
    }

    public function setPermissions($permissions) {
        $this->permissions = $permissions;
    }
}

?>